apps/condo/domains/ticket/access/Ticket.js

```diff
@@ -31,49 +31,130 @@ async function canReadTickets ({ authentication: { item: user }, context }) {
         }
     }
 
-    const userDivisionsInfo = await getUserDivisionsInfo(context, user.id)
+    const userEmployees = await find('OrganizationEmployee', {
+        user: { id: user.id },
+        deletedAt: null,
+    })
+    const userEmployeeRoles = await find('OrganizationEmployeeRole', {
+        id_in: userEmployees.map(employee => employee.role),
+    })
+    const employeeToVisibilityType = userEmployees.map(employee => {
+        const visibilityType = userEmployeeRoles.find(role => role.id === employee.role).ticketVisibilityType
+        return { employee: employee, visibilityType }
+    })
+
+    const organizationTicketVisibilityOrganizationIds = []
+    const propertiesVisibilityPropertyIds = []
+    const propertyAndSpecializationVisibilitySearchStatement = []
+
+    const employeeWithPropertyVisibilityType = mapEmployeeToVisibilityTypeToEmployees(employeeToVisibilityType, PROPERTY_TICKET_VISIBILITY)
+    const employeeWithPropertyAndSpecializationVisibilityType = mapEmployeeToVisibilityTypeToEmployees(employeeToVisibilityType, PROPERTY_AND_SPECIALIZATION_VISIBILITY)
+
+    const employeeWithPropertyVisibilityIds = [...employeeWithPropertyVisibilityType, ...employeeWithPropertyAndSpecializationVisibilityType].map(employee => employee.id)
+    let propertyScopes = []
+    if (employeeWithPropertyVisibilityIds.length > 0) {
+        propertyScopes = await getPropertyScopes(employeeWithPropertyVisibilityIds)
+    }
```

Comment:
It looks a little complex. Really you need to get all employees and split them into three groups. Then for each group, you need to get some extra data for the query. Try to simplify this logic. May be you need to extract some util for that

Reply:
Yes, but `property` and `propertyAndAssignee` groups have corner cases, like:
if employee with property visibility added in property scope which includes all properties, then he can read all tickets in organization (= all properties in organization).

Moved access queries calculation in util